### PR TITLE
SSO: Add custom message to new user email

### PR DESCRIPTION
**Pull request by agrullon95**
Fixes https://github.com/Automattic/wp-calypso/issues/86760

## Proposed changes:

* Adds a "Custom Message" field to the wp-admin new user form

**New user form:**

image

**Email:**
CleanShot 2024-02-01 at 10 40 34@2x

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

1. On an Atomic, JN, or whatever non-simple site, install Jetpack.
2. Make sure SSO is on, it should be on by default.
3. Sync this PR to your site.
4. Navigate to `/wp-admin/user-new.php`
5. Add a username, email, and custom message.
6. Verify the custom message is sent in email.

### CODE REVIEW

**Comment by crisbusquets** [comment]
Great!

Works for me, @agrullon95 👍 

Add-New-User-‹-My-WordPress-Site-—-WordPress

Inbox-phoenixowl777-proton-me-Proton-Mail

File: projects/plugins/jetpack/modules/sso.php [hunk 1]
```diff
@@ -563,6 +598,19 @@ public function jetpack_user_table_styles() {
...
+	 */
+	public function jetpack_users_new_form_styles() {
+		?>
+		<style>
+			#createuser .form-field textarea {
+				width: 25em;
+			}
+		</style>
+		<?php
+	}
```
Comment (escapemanuele) [inline projects/plugins/jetpack/modules/sso.php]: This works! I wonder if we could leverage `jetpack-sso-admin-create-user.css`, enqueued by `render_invitation_email_message` when in `user-new`? 

Maybe with a different wording we could use that function to add the Notice and enqueue all the CSS for that page.

Comment (agrullon95) [reply projects/plugins/jetpack/modules/sso.php]: @escapemanuele - That's a good idea. We can leverage that CSS file. I'll move my styling into `jetpack-sso-admin-create-user.css`.

I think it makes more sense to use `add_action( 'admin_print_styles-user-new.php', array( $this, 'jetpack_users_new_form_styles' ) );`, move my styling into `jetpack-sso-admin-create-user.css`, then enqueue the CSS in that function (function `jetpack_users_new_form_styles`).

This way we can separate the styling and the notice logic.

 WDYT?

Comment (escapemanuele) [reply projects/plugins/jetpack/modules/sso.php]: That makes totally sense for me @agrullon95 👍

**Review by escapemanuele — APPROVED** [review]
Thanks for addressing my feedback, looks great now!

**Comment by alshakero** [comment]
IMHO, this message field seems like an attribute of the user when it's not. Remember, the point of this project is gaining the trust of people who are used to WP Core. This field doesn't exist in Core and will be confusing unless explained well.

Let's give it a placeholder, something like:

"This user will be invited to WordPress.com, you can include a customized welcoming message that we'll make sure to pass along with the invitation".

**Comment by escapemanuele** [comment]
> Let's give it a placeholder,

Makes total sense 👍

**Comment by escapemanuele** [comment]
Placeholder works!

image

Would it be better if we change the color to looks a little less like text and more like a placeholder? @agrullon95

**Comment by crisbusquets** [comment]
> Placeholder works!
> 
> image
> 
> Would it be better if we change the color to looks a little less like text and more like a placeholder? @agrullon95

Hi!

We can do this instead of a placeholder. This way the message is always visible, even if they edit the textarea. WDYT?

d_wpadmin_invite_message

Figma file: DASWajWXHUUjF4pu1yo4fI-fi-246_1228

**Comment by alshakero** [comment]
I like [this](https://github.com/Automattic/jetpack/pull/35395#issuecomment-1926485104) more than the placeholder for sure. I was a bit reluctant to suggest the placeholder idea because it misuses the field.

**Comment by alshakero** [comment]
Pushed, beautiful now
image

**Pull request merged**